native_client/wasm/bindings.cc

```diff
@@ -0,0 +1,352 @@
+        return TokenMetadataStub{
+            tokenMetadata.text,
+            tokenMetadata.timestep,
+            tokenMetadata.start_time
+        };
+    }
+};
+
+typedef struct CandidateTranscriptStub {
+    std::vector<TokenMetadataStub> tokens;
+    double confidence;
+
+    static CandidateTranscriptStub fromCandidateTranscript(CandidateTranscript candidateTranscript) {
+        std::cout << "Converting from CandidateTranscript" << std::endl;
+        std::vector<TokenMetadataStub> tokens = std::vector<TokenMetadataStub>(candidateTranscript.num_tokens);
+
+        for (int i = 0; i < candidateTranscript.num_tokens; i++) {
+            const TokenMetadata candidateToken = candidateTranscript.tokens[i];
+            TokenMetadataStub token = TokenMetadataStub::fromTokenMetadata(candidateToken);
+            tokens[i] = token;
+        }
+
+
+        return CandidateTranscriptStub{
+            tokens,
+            candidateTranscript.confidence
+        };
+    }
+};
+
+typedef struct MetadataStub {
+    std::vector<CandidateTranscriptStub> transcripts;
+
+    static MetadataStub fromMetadata(Metadata* metadata) {
+        std::cout << "Converting from Metadata" << std::endl;
+        std::cout << "Number of transcripts: " << metadata->num_transcripts << std::endl;
+
+        std::vector<CandidateTranscriptStub> transcripts = std::vector<CandidateTranscriptStub>(metadata->num_transcripts);
+        for (int i = 0; i < metadata->num_transcripts; i++) {
+            std::cout << "Converting transcript " << i << std::endl;
+            const CandidateTranscript candidateTranscript = metadata->transcripts[i];
+            std::cout << "Transcript confidence " << candidateTranscript.confidence << std::endl;
+            CandidateTranscriptStub transcript = CandidateTranscriptStub::fromCandidateTranscript(candidateTranscript);
+            std::cout << "Converted transcript confidence " << transcript.confidence << std::endl;
+            transcripts[i] = transcript;
+        }
+
+        return MetadataStub{
+            transcripts
+        };
+    }
+} MetadataStub;
+
+class Stream {
+ public:
+  Stream(StreamingState* streamingState)
+    : streamingState(streamingState) {}
+
+  void feedAudioContent(std::vector<short> audioBuffer) {
+    STT_FeedAudioContent(this->streamingState, audioBuffer.data(), audioBuffer.size());
+  }
+
+  std::string intermediateDecode() {
+    char* tempResult = STT_IntermediateDecode(this->streamingState);
+    if (!tempResult) {
+      // There was some error, return an empty string.
+      return std::string();
+    }
+
+    // We must manually free the string if something was returned to us.
+    std::string result = tempResult;
+    STT_FreeString(tempResult);
+    return result;
+  }
+
+  MetadataStub intermediateDecodeWithMetadata(unsigned int numResults = 1) {
+    Metadata* tempResult =
+      STT_IntermediateDecodeWithMetadata(this->streamingState, numResults);
+    if (!tempResult) {
```

Comment:
Some mixed indentation in this file. If there's tabs and spaces, please normalize to spaces everywhere.